zk/stages/stage_sequence_execute.go

```diff
@@ -164,51 +170,85 @@ func SpawnSequencingStage(
 		defer tx.Rollback()
 	}
 
+	hermezDb, err := hermez_db.NewHermezDb(tx)
+	if err != nil {
+		return err
+	}
+
 	executionAt, err := s.ExecutionAt(tx)
 	if err != nil {
 		return err
 	}
-	nextBlockNum := executionAt + 1
 
 	parentBlock, err := rawdb.ReadBlockByNumber(tx, executionAt)
 	if err != nil {
 		return err
 	}
 
-	hermezDb, err := hermez_db.NewHermezDb(tx)
+	lastBatch, err := stages.GetStageProgress(tx, stages.HighestSeenBatchNumber)
 	if err != nil {
 		return err
 	}
 
-	/*
-		batch: here we need to open up a new batch ready to populate with transactions.
-
-		the loop below that yields the `best` transactions from the pool should loop until we hit one of two
-		limits:
-		1: time based - for the batch to be closed
-		2: counter based - if the transactions in the batch have exhausted the virtual counters
+	nextBlockNum := executionAt + 1
+	thisBatch := lastBatch + 1
+	newBlockTimestamp := uint64(time.Now().Unix())
 
-		It can work with the open mining block whilst all of this is happening until we hit one of those conditions.
+	header := &types.Header{
+		ParentHash: parentBlock.Hash(),
+		Coinbase:   constMiner,
+		Difficulty: blockDifficulty,
+		Number:     new(big.Int).SetUint64(nextBlockNum),
+		GasLimit:   blockGasLimit,
+		Time:       newBlockTimestamp,
+	}
 
-		After the condition is met we need to close the batch and hold it in the db for a future stage to submit this
-		for proving
-	*/
+	stateReader := state.NewPlainStateReader(tx)
+	ibs := state.New(stateReader)
 
-	lastBatch, err := stages.GetStageProgress(tx, stages.HighestSeenBatchNumber)
-	if err != nil {
-		return err
+	// here we have a special case and need to inject in the initial batch on the network before
+	// we can continue accepting transactions from the pool
+	if executionAt == 0 {
+		injected, err := hermezDb.GetL1InjectedBatch(0)
```

Comment:
maybe we can extract this into a function like `injectInitialBatch` to clean-up the logic and make it more readable?